### DISCUSSION TITLE
Are native controls used on all platforms except web?

### DISCUSSION THREAD

**Discussion by mrc-eu** [unoplatform/uno#13733] [General]
Until yesterday I always thought that Uno draws all the controls, meaning there not using the native controls. I also read an article about embedding native controls, so that assumes that by default Uno does not use them.

Yesterday I saw this video from Nick Randolph:
https://youtu.be/0zvzfCGO0gI?t=424

The link starts at the point where he explains the difference between the different SDK's. There he somewhat says, if I understand it correctly, that Uno uses the native controls but changes it looks so that it will look the same on all platforms.

So, where Flutter does draw all the controls on a canvas, he claims that Uno does not. Is that correct?

**Comment by jeromelaban** [comment]
Thanks for the question!

Indeed at this time, on iOS, Android, mac Catalys and Web, Uno Platform use a small number of platform primitives that are difficult to replace or replicate, such as UITextView on iOS, EditView on Android or `Input` on Web, for which the "chrome" is removed. The rest of the controls are drawn using platform primitives, but in an integrated way that allows easy and performant integration of native controls, such as in our latest [MAUI Embedding support](https://aka.platform.uno/maui-embedding).

Uno Platform is doing this for multiple reasons, but one of the prominent reason is Accessibility and the ability to take advantage of platform features (such IME).

For other targets, which are using a Skia backend (Wpf, Gtk and Framebuffer), Uno is currently rendering everything per the pixel on a canvas, with some exceptions that we're currently working on (https://github.com/unoplatform/uno/issues/9417).
